### PR TITLE
Improve markdown table's styling

### PR DESCRIPTION
**Pull request by budnix**
### Context

The PR improves the markdown table styling. The original [PR](https://github.com/handsontable/handsontable/pull/8341) adds a `word-break: break-word;` style to the table element. That fixed the table responsiveness for the migration guides page but it worsened the table appearance on other sites.

The PR fixes that by applying the above-mentioned style only for migration guides pages. 

Before fix:
image

After fix:
image

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1.
2.
3.

### Affected project(s):
- [ ] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: docs/.vuepress/theme/styles/index.styl [hunk 1]
```diff
@@ -177,7 +177,6 @@ a {
 table {
   width: 100%;
   display: table;
-  word-break: break-word;
```
Comment (Siemienik) [inline docs/.vuepress/theme/styles/index.styl:180]: It is required for responsiveness: see  #8296, #8341 

Should we attempt to turn it off only on mobile? Have any idea?

Comment (budnix) [reply docs/.vuepress/theme/styles/index.styl:180]: From my observation, only very wide tables look ugly. That's why I decided to style the table differently only for guides pages.

Comment (Siemienik) [reply docs/.vuepress/theme/styles/index.styl:180]: how about tables listed by @aninde ? 
https://github.com/handsontable/handsontable/issues/8296

Comment (budnix) [reply docs/.vuepress/theme/styles/index.styl:180]: Thanks! I've checked that and improved the style by adding media queries for narrow viewports c0fe37bbd1e76f9ae40ec1eb7a5818ffd41abeff. I think that now, all or at least most of the cases are covered.

**Review by Siemienik — APPROVED** [review]
LGTM

**Pull request merged**